### PR TITLE
report terraform progress when creating cluster.

### PR DESCRIPTION
**Pull request by abhinavdahiya**
/cc @crawford @wking

### CODE REVIEW

**Comment by crawford** [comment]
I don't see any new output with this patch applied. Do I have to do anything special?

**Comment by abhinavdahiya** [comment]
> I don't see any new output with this patch applied. Do I have to do anything special?

Progress report is supressed, so libvirt shows no progress... on aws so can see the progress being reported.

using aws:
```
INFO Consuming "Install Config" from target directory
INFO Creating cluster...
INFO Applying resources for cluster, so far 13 created
INFO Applying resources for cluster, so far 40 created
INFO Applying resources for cluster, so far 56 created
INFO Applying resources for cluster, so far 73 created
INFO Applying resources for cluster, so far 84 created
INFO Applying resources for cluster, so far 95 created
INFO Applying resources for cluster, so far 119 created
INFO Applying resources for cluster, so far 131 created
INFO Applying resources for cluster, so far 142 created
INFO Applying resources for cluster, so far 153 created
INFO Waiting up to 30m0s for the Kubernetes API...
INFO API v1.11.0+ad103ed up
INFO Waiting up to 30m0s for the bootstrap-complete event...
ERROR: logging before flag.Parse: E0109 10:15:16.504934   30285 streamwatcher.go:109] Unable to decode an event from the watch stream: http2: server sent GOAWAY and closed the connection; LastStreamID=3, ErrCode=NO_ERROR, debug=""
WARNING RetryWatcher - getting event failed! Re-creating the watcher. Last RV: 2677
WARNING Failed to connect events watcher: Get https://adahiya-0-api.coreservices.team.coreos.systems:6443/api/v1/namespaces/kube-system/events?resourceVersion=2677&watch=true: dial tcp 3.89.82.220:6443: connect: connection refused
WARNING Failed to connect events watcher: Get https://adahiya-0-api.coreservices.team.coreos.systems:6443/api/v1/namespaces/kube-system/events?resourceVersion=2677&watch=true: dial tcp 3.89.82.237:6443: connect: connection refused
WARNING Failed to connect events watcher: Get https://adahiya-0-api.coreservices.team.coreos.systems:6443/api/v1/namespaces/kube-system/events?resourceVersion=2677&watch=true: dial tcp 3.89.82.220:6443: connect: connection refused
WARNING Failed to connect events watcher: Get https://adahiya-0-api.coreservices.team.coreos.systems:6443/api/v1/namespaces/kube-system/events?resourceVersion=2677&watch=true: dial tcp 3.89.82.182:6443: connect: connection refused
WARNING Failed to connect events watcher: Get https://adahiya-0-api.coreservices.team.coreos.systems:6443/api/v1/namespaces/kube-system/events?resourceVersion=2677&watch=true: dial tcp 3.89.82.130:6443: i/o timeout
WARNING Failed to connect events watcher: Get https://adahiya-0-api.coreservices.team.coreos.systems:6443/api/v1/namespaces/kube-system/events?resourceVersion=2677&watch=true: dial tcp 3.89.82.195:6443: connect: connection refused
INFO Destroying the bootstrap resources...
INFO Waiting up to 10m0s for the openshift-console route to be created...
INFO Install complete!
```

using libvirt:
```
INFO Consuming "Install Config" from target directory
INFO Creating cluster...
INFO Waiting up to 30m0s for the Kubernetes API...
INFO API v1.11.0+ad103ed up
INFO Waiting up to 30m0s for the bootstrap-complete event...
INFO Destroying the bootstrap resources...
INFO Waiting up to 10m0s for the openshift-console route to be created...
INFO Install complete!
```

File: pkg/terraform/terraform.go [hunk 1]
```diff
@@ -51,12 +52,32 @@ func Apply(dir string, platform string, extraArgs ...string) (path string, err e
...
+	isTerminal := terminal.IsTerminal(int(os.Stdout.Fd()))
+	progressCh := make(chan texec.Progress)
+	var prevP texec.Progress
+	defer close(progressCh)
+	go func(progressCh <-chan texec.Progress) {
+		for p := range progressCh {
+			if shouldReportProgress(prevP, p) {
+				prevP = p
+				if isTerminal {
+					logrus.Infof("Applying resources for cluster, so far %s", p)
```
[the remark this answers is not in the document]
Comment (abhinavdahiya) [reply pkg/terraform/terraform.go]: my thinking was
* I don't think we want to report progress if user sets log-level to warning
* I don't want to show these when it is not being used by a human.

So i thought we can just print these using logrus.

**Comment by abhinavdahiya** [comment]
@crawford updated to provide progress using a progress bar.

**Comment by abhinavdahiya** [comment]
Oh, don't know how to provide example here...
maybe this https://asciinema.org/a/Op0ZEi2Ljm0nObrvUeX79aaOV?t=41

**Pull request closed without merging**